lib/components/molecules/Pagination/Pagination.js

```diff
@@ -0,0 +1,253 @@
+
+    if (totalPages > totalBlocks) {
+      const startPage = Math.max(2, activePage - pageNeighbours);
+      const endPage = Math.min(totalPages - 1, activePage + pageNeighbours);
+
+      let pages = pageRange(startPage, endPage);
+
+      /**
+       * hasLeftSpill: has hidden pages to the left
+       * hasRightSpill: has hidden pages to the right
+       * spillOffset: number of hidden pages either to the left or to the right
+       */
+      const hasLeftSpill = startPage > 2;
+      const hasRightSpill = totalPages - endPage > 1;
+      const spillOffset = totalNumbers - (pages.length + 1);
+
+      switch (true) {
+        // handle: (1) < {5 6} [7] {8 9} (10)
+        case hasLeftSpill && !hasRightSpill: {
+          const extraPages = pageRange(startPage - spillOffset, startPage - 1);
+          pages = [LEFT_PAGE, ...extraPages, ...pages];
+          break;
+        }
+
+        // handle: (1) {2 3} [4] {5 6} > (10)
+        case !hasLeftSpill && hasRightSpill: {
+          const extraPages = pageRange(endPage + 1, endPage + spillOffset);
+          pages = [...pages, ...extraPages, RIGHT_PAGE];
+          break;
+        }
+
+        // handle: (1) < {4 5} [6] {7 8} > (10)
+        case hasLeftSpill && hasRightSpill:
+        default: {
+          pages = [LEFT_PAGE, ...pages, RIGHT_PAGE];
+          break;
+        }
+      }
+
+      return [1, ...pages, totalPages];
+    }
+
+    return pageRange(1, totalPages);
+  };
+
+  useEffect(() => {
+    setDisabledFirstNdPrevious(activePage === 1);
+    setDisabledNextAndLast(activePage === pageNumbers.length);
+  }, [activePage]);
+
+  const getCurrentDataSet = () => {
+    const indexOfLastPost = activePage * pageSize;
+    const indexOfFirstPost = indexOfLastPost - pageSize;
+    return data && data.length && data.slice(indexOfFirstPost, indexOfLastPost);
+  };
+
+  const executePagination = page => {
+    setActivePage(page);
+    if (typeof onPageChange === 'function') {
+      onPageChange(page);
+    }
+  };
+
+  const handleForwardClick = () => {
+    const nextPage = activePage + 1;
+    executePagination(nextPage);
+  };
+
+  const handleBackwardClick = () => {
+    const previousPage = activePage - 1;
+    executePagination(previousPage);
+  };
+
+  const jumpToFirst = () => {
+    const firstPage = 1;
+    executePagination(firstPage);
+  };
+
+  const jumpToLast = () => {
+    const lastPage = pageNumbers.length;
+    executePagination(lastPage);
+  };
+
+  const renderDataComponent = () => {
+    if (typeof renderPaginationComponent === 'function') {
+      return renderPaginationComponent({
+        data: getCurrentDataSet(),
+        activePage,
+      });
+    }
+    return false;
+  };
+
+  const pages = fetchPageNumbers();
+
+  return (
+    <>
+      {renderDataComponent()}
+      <nav className={className} arial-label={paginationAriaLabel}>
+        <ul className="pagination">
+          {showFirstBtn && (
+            <li className="first">
```

Comment:
When you already have first and last button here why does <PaginationList>  has code for adding

Reply:
Done